src/client/pythonEnvironments/common/environmentIdentifier.ts

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import * as fsapi from 'fs-extra';
+import * as path from 'path';
+import { traceWarning } from '../../common/logger';
+import { createDeferred } from '../../common/utils/async';
+import { getEnvironmentVariable } from '../../common/utils/platform';
+import { EnvironmentType } from '../info';
+
+function pathExists(absPath: string): Promise<boolean> {
+    const deferred = createDeferred<boolean>();
+    fsapi.exists(absPath, (result) => {
+        deferred.resolve(result);
+    });
+    return deferred.promise;
+}
+
+function or(...arr: boolean[]): boolean {
+    return arr.filter((x) => x).length > 0;
```

Comment:
```suggestion
    return arr.includes(true);
```
this is like, the nitpickiest of nitpickings

Reply:
This is better.